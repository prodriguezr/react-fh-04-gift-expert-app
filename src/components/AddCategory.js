import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handlerOnChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handlerOnSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length === 0) return;

    setCategories((oldCategories) => [inputValue, ...oldCategories]);

    setInputValue('');
  };

  return (
    <form onSubmit={handlerOnSubmit}>
      <input type='text' value={inputValue} onChange={handlerOnChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
