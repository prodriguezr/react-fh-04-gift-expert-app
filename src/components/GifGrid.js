import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from '../hooks';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p className='animate__flash'>Loading...</p>}

      <div className='card-grid'>
        {images.map(({ id, title, url }) => (
          <GifGridItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
