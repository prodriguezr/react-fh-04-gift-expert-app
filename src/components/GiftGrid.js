import { GiftGridItem } from './GiftGridItem';

import { useFetchGifts } from '../hooks';

export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifts(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p className='animate__flash'>Loading...</p>}

      <div className='card-grid'>
        {images.map(({ id, title, url }) => (
          <GiftGridItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};
