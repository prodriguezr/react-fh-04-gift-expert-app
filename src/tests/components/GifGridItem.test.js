import { shallow } from 'enzyme';
import { GifGridItem } from '../../components';

describe('Tests on GifGridItem', () => {
  const title = 'Test Title';
  const url = 'http://test.cl/img.png';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('It should display the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('It must have a paragraph with the title', () => {
    expect(wrapper.find('p').text().trim()).toBe(title);
  });

  test('The src property of the image must be equal to the url property', () => {
    expect(wrapper.find('img').prop('src')).toBe(url);
  });

  test('The alt property of the image must be equal to the title property', () => {
    expect(wrapper.find('img').prop('alt')).toBe(title);
  });

  test('Must contain animate__fadeIn css class', () => {
    const css = 'animate__fadeIn';
    const div = wrapper.find('div');

    expect(div.prop('className').includes(css)).toBe(true);
  });
});
