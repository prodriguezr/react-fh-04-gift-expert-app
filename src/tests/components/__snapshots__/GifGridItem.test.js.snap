// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGridItem It should display the component correctly 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Test Title"
    src="http://test.cl/img.png"
  />
  <p>
    Test Title
  </p>
</div>
`;
